test/test_mutable_list.jl

```diff
@@ -110,12 +131,32 @@
                 @testset "map" begin
                     l = MutableLinkedList{Int}(1:n...)
                     @test map(x -> 2x, l) == MutableLinkedList{Int}(2:2:2n...)
+                    l2 = MutableLinkedList{Float64}()
+                    @test map(x -> x*im, l2) == MutableLinkedList{Complex{Float64}}()
+                    @test map(Int32, l2) == MutableLinkedList{Int32}()
+                    f(x) = x % 2 == 0 ? convert(Int8, x) : convert(Float16, x)
+                    @test typeof(map(f, l)) == MutableLinkedList{Real}
                 end
 
                 @testset "filter" begin
                     l = MutableLinkedList{Int}(1:n...)
                     @test filter(x -> x % 2 == 0, l) == MutableLinkedList{Int}(2:2:n...)
                 end
+
+                @testset "show" begin
+                    l = MutableLinkedList{Int32}(1:n...)
+                    let
+                        io = IOBuffer()
+                        @test sprint(io -> show(io, l.node.next)) == "$(typeof(l.node.next))($(l.node.next.data))"
+                        begin io1 = IOBuffer()
+                              write(io1, "MutableLinkedList{Int32}(");
+                              write(io1, join(l, ", "));
+                              write(io1, ")")
+                              io1.ptr = 1
+                              @test sprint(io -> show(io, l)) == read(io1, String)
```

Comment:
Thanks for adding the tests. 

To seek back to the beginning of an IOBuffee, it's generally better to use `seekstart()`, rather than seeing `ptr`directly.

Otherwise, looks go to me!